### PR TITLE
CRIMAPP-1002-1003-1004-1005-1006 Partner employment loop forms + Partner self assessment tax bill forms

### PR DESCRIPTION
**Pull request by ripan**
## Description of change
- Add partner employment loop forms
- Add self assessment tax bill form

## Link to relevant ticket

What is the partner's employment status?
https://dsdmoj.atlassian.net/browse/CRIMAPP-1000

Who does the partner work for
https://dsdmoj.atlassian.net/browse/CRIMAPP-1002

The partner's employment details
https://dsdmoj.atlassian.net/browse/CRIMAPP-1003

Deductions from the partner's pay
https://dsdmoj.atlassian.net/browse/CRIMAPP-1004

Shopping basket for Employed Partner
https://dsdmoj.atlassian.net/browse/CRIMAPP-1005

Does the partner pay a self-assessment tax bill received in the last 2 years?
https://dsdmoj.atlassian.net/browse/CRIMAPP-1006

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -50,7 +50,7 @@ gem 'laa-criminal-applications-datastore-api-client',
 
 gem 'laa-criminal-legal-aid-schemas',
     github: 'ministryofjustice/laa-criminal-legal-aid-schemas',
-    tag: 'v1.1.15'
+    ref: '6a59d8ac5f00579bb7f536da2e1dfba4b0577e1b'
```
Comment (RoseSAK) [inline Gemfile:53]: Is this right?

**Review by RoseSAK — APPROVED** [review]
Looks good, only one comment

File: Gemfile (hunk 1 above)
Comment (ripan) [reply Gemfile]: @RoseSAK Updated schema version to v1.1.16

File: app/services/decisions/income_decision_tree.rb [hunk 2]
```diff
@@ -128,38 +148,73 @@ def after_partner_income_benefits
...
+    end
+
+    def start_partner_employment_journey # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+      case form_object.partner_employment_status
+      when [EmploymentStatus::EMPLOYED.to_s]
+        if FeatureFlags.employment_journey.enabled?
+          edit('/steps/income/partner/employment_income')
+        else
+          show(:employed_exit)
+        end
```
Comment (hiboabd) [inline app/services/decisions/income_decision_tree.rb:181]: I think the [routing here](https://www.figma.com/design/thkvnDBbQbHlSqzt1TE3lk/Crime-Apply?node-id=14214-72385&t=w4rSIOQHnzonFeBv-0) needs amending, but this pr is quite big. Do you want to address in a separate pr maybe?

**Review by hiboabd — COMMENTED** [review]
Looks good, i just caught a few things. I think the error validation of each form could do with double checking 
Screenshot 2024-06-21 at 15 47 54

Screenshot 2024-06-21 at 15 47 25

File: app/forms/steps/income/partner/employment_income_form.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
+module Steps
+  module Income
+    module Partner
+      class EmploymentIncomeForm < Steps::BaseFormObject
+        attribute :amount, :pence
```
Comment (hiboabd) [inline app/forms/steps/income/partner/employment_income_form.rb:5]: Adding employment payments is breaking the view of income payments 
Screenshot 2024-06-21 at 16 17 39

File: config/locales/en/helpers.yml [hunk 4]
```diff
@@ -615,15 +640,43 @@ en:
...
+        deduction_attributes:
+          amount: What is their salary or wage?
+        income_tax:
+          amount: *amount_label
+          frequency_options: *frequency_options
+        national_insurance:
+          amount: *amount_label
+          frequency_options: *frequency_options
+        other:
+          amount: Enter total amount of all other deductions
```
[the remark this answers is not in the document]
Comment (ripan) [reply config/locales/en/helpers.yml:659]: Not required as per the design

File: config/locales/en/helpers.yml [hunk 5]
```diff
@@ -615,15 +640,43 @@ en:
...
+          frequency_options: *frequency_options
+          details: Enter details of other deductions
```
[the remark this answers is not in the document]
Comment (ripan) [reply config/locales/en/helpers.yml:661]: Yes

File: spec/services/decisions/income_decision_tree_spec.rb [hunk 6]
```diff
@@ -183,12 +196,36 @@
...
+    #   let(:partner_employment_status) { [EmploymentStatus::EMPLOYED.to_s] }
+    #
+    #   before do
+    #     allow(form_object).to receive(:partner_employment_status).and_return([EmploymentStatus::EMPLOYED.to_s])
+    #   end
+    #
+    #   it 'redirects to the `employed_exit` page' do
+    #     expect(subject).to have_destination(:employed_exit, :show, id: crime_application)
+    #   end
+    # end
```
[the remark this answers is not in the document]
Comment (ripan) [reply spec/services/decisions/income_decision_tree_spec.rb:226]: No. Deleted

File: spec/services/decisions/income_decision_tree_spec.rb [hunk 7]
```diff
@@ -213,6 +252,63 @@
...
+    #   let(:partner_employment_status) { [EmploymentStatus::SELF_EMPLOYED.to_s] }
+    #
+    #   before do
+    #     allow(form_object).to receive(:partner_employment_status).and_return([EmploymentStatus::SELF_EMPLOYED.to_s])
+    #   end
+    #
+    #   it 'redirects to the `employed_exit` page' do
+    #     expect(subject).to have_destination(:employed_exit, :show, id: crime_application)
+    #   end
+    # end
```
[the remark this answers is not in the document]
Comment (ripan) [reply spec/services/decisions/income_decision_tree_spec.rb:265]: No. Deleted

File: app/services/decisions/income_decision_tree.rb (hunk 2 above)
Comment (ripan) [reply app/services/decisions/income_decision_tree.rb:181]: Yes, i will handle in separate PR

File: config/locales/en/errors.yml [hunk 8]
```diff
@@ -1002,10 +1036,41 @@ en:
...
+            frequency:
+              blank: Select how often they get this payment
+              inclusion: Select how often they get this payment
+          summary:
+            job_title:
+              blank: Enter their job title
+            amount:
+              blank: Enter their salary or wage
+              not_a_number: Enter their salary or wage
+              greater_than: Enter their salary or wage greater than 0
```
[the remark this answers is not in the document]
Comment (ripan) [reply config/locales/en/errors.yml]: Fixed the content

File: config/locales/en/errors.yml [hunk 9]
```diff
@@ -1029,3 +1094,26 @@ en:
...
+            amount:
+              blank: Enter amount
+              not_a_number: Enter amount
+              greater_than: Enter amount greater than 0
+            frequency:
+              blank: Select how often the amount gets deducted
+              inclusion: Select how often the amount gets deducted
+            details:
+              blank: Enter details
+              invalid: Details not required
```
[the remark this answers is not in the document]
Comment (ripan) [reply config/locales/en/errors.yml:1108]: Not sure/remember. lets keep it as its not doing any harm :)

File: app/forms/steps/income/partner/employment_income_form.rb (hunk 3 above)
Comment (ripan) [reply app/forms/steps/income/partner/employment_income_form.rb:5]: Will handle in separate bug ticket

**Comment by ripan** [comment]
I will fix the following(as per your review) in a separate bug tickets
- https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/pull/946#pullrequestreview-2132779800
- https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/pull/946#discussion_r1649108241
